### ISSUE TITLE
Search input field sometimes not being clickable in Safari (iOS & macOS) when using oscar.

### ISSUE THREAD

**Issue by tiekoetter** [searxng/searxng#323]
The search bar sometimes gets blocked after hitting enter or pressing the search button. This happens in Safari 14.1.2 (16611.3.10.1.6) when using the oscar theme with pointhi, logicodev or logicodev-dark with all recent versions of searx or searxng.

Example video showing the bug: https://youtu.be/psY2jXM-vUE

I already tried to fix this (https://github.com/tiekoetter/searxng/commit/c45611e5c6583ca01864c7c2ed8ae51bf7c3e6f3) but I am not a developer so there is probably a better solution.

Example video showing the expected behavior using my fix: https://youtu.be/FqNf_DmK68E

**Comment by dalf** [comment]
Thank you for the details and the commit.

I've opened a pull request which _should_ fix the issue: https://github.com/searxng/searxng/pull/330 ( https://github.com/dalf/searxng/tree/fix-323 )

_should_ because I don't have Safari, so I can't test it. I can only see there is no visual change on Firefox and Chrome with this PR.

About https://github.com/tiekoetter/searxng/commit/c45611e5c6583ca01864c7c2ed8ae51bf7c3e6f3 :

```patch
--- a/searx/static/themes/oscar/css/bootstrap.css
+++ b/searx/static/themes/oscar/css/bootstrap.css
@@ -3888,7 +3888,7 @@ tbody.collapse.in {
   margin-bottom: 0;
 }
 .input-group .form-control:focus {
-  z-index: 3;
+  z-index: 2;
 }
 .input-group-lg > .form-control,
 .input-group-lg > .input-group-addon,
```

in the pull request it is replaced by:

```patch
diff --git a/searx/static/themes/oscar/src/less/bootstrap/typeahead.less b/searx/static/themes/oscar/src/less/bootstrap/typeahead.less
index de1c6ef05..7513da6a2 100644
--- a/searx/static/themes/oscar/src/less/bootstrap/typeahead.less
+++ b/searx/static/themes/oscar/src/less/bootstrap/typeahead.less
@@ -93,6 +93,9 @@
     z-index: 2;
     //disabled status
     //overwrite inline styles of .tt-query
+    &:focus {
+      z-index: 2;
+    }
     &[disabled],
     &[readonly],
     fieldset[disabled] & {
```

it applies the z-index on the element HTML, but it avoids to change bootstrap CSS classes.

---

About 

```patch
--- a/searx/static/themes/oscar/css/bootstrap.css
+++ b/searx/static/themes/oscar/css/bootstrap.css
@@ -4026,7 +4026,6 @@ select[multiple].input-group-sm > .input-group-btn > .btn {
 .input-group-btn:last-child > .btn,
 .input-group-btn:last-child > .btn-group {
   z-index: 2;
-  margin-left: -1px;
 }
```

I can see where the oscar theme uses `btn-group`.

**Comment by tiekoetter** [comment]
I removed the btn-group margin-left because the right side of the white border, when clicking on the search bar with logicodev-dark, gets cutoff.

**Comment by dalf** [comment]
Can you post a screenshot?

**Comment by tiekoetter** [comment]
Without the removal of margin-left:
![image](https://user-images.githubusercontent.com/77782671/133885905-b2bd9308-7fdd-480e-babe-0b61a4c55cd1.jpeg)

With the removal of margin-left:
![image](https://user-images.githubusercontent.com/77782671/133885928-d8aa015c-aa4d-435a-b7bc-84cba939458d.jpeg)

**Comment by dalf** [comment]
Thank you.
Now I understand why there is `z-index: 3;` by default: when focused, the input text element is above the others, so the white border is visible.

What I don't understand if why with `z-index: 3` makes it disappear in Safari.

When `margin-left: -1px;` is removed, the layout on Firefox slightly change:
* Without `margin-left: -1px;`:
![image](https://user-images.githubusercontent.com/1594191/133888270-a6b3e54d-7fe0-43d4-9d19-4990c986ef93.png)
* With `margin-left: -1px;`:
![image](https://user-images.githubusercontent.com/1594191/133888277-dd34e511-1593-440a-a9a5-5fbb19dc10cd.png)

**Comment by dalf** [comment]
tiekoetter I've merged #330 
The code is different from the original pull request: the changes are enabled only for Safary using CSS hack:
* the layout is not changed with FF / Chrome / etc...
* the layout is fixed with Opera (using the change you have suggested).

I don't like CSS hack, but I don't know how to fix the issue without having Safari ( see https://github.com/searxng/searxng/pull/330#issuecomment-922530471 ). Feel free to reopen.
